agent/stats/types.go

```diff
@@ -29,10 +29,18 @@ type ContainerStats struct {
 	memoryUsage       uint64
 	storageReadBytes  uint64
 	storageWriteBytes uint64
+	restartCount      *int64
```

Comment:
Is there any specific reason this (and `UsageStats.RestartCount` on L65) needs to be of type `*int64`? Can it be of type `uint64` like the struct fields above it?

Reply:
yes! the reason it's int instead uint is because the count comes back from the RestartTracker.GetRestartCount function as an int.

The reason it's a pointer is because we want to exclude this metric if the counter doesn't exist, so having it be a pointer tells us if the value is unset, in which case we don't want to send the RestartStatsSet object to TCS.

Reply:
se https://github.com/aws/amazon-ecs-agent/pull/4196/files#diff-020a54a32e6f08ac86c2901d77f7ca89b0935ddf7e974f42f7ab0f5e353a04c9R242

Reply:
Makes sense, thanks!